AGENTS.md

```diff
@@ -350,18 +383,87 @@ def teardown_method(self) -> None:
+
+**Important**: Only test the specific features and behaviors that our codebase actually uses from the external library.
+
+#### Contract Test Structure
+
 ```python
 # tests/contract/test_github_api.py
+"""Contract tests for GitHub API to ensure API structure matches expectations."""
+import pytest
+
 class TestGitHubAPIContract:
+    """Validate GitHub API endpoint contracts."""
+    
     def test_repository_endpoint_structure(self) -> None:
-        """Ensure GitHub API returns expected structure."""
+        """Ensure GitHub API returns expected repository structure."""
         response = github_api.get_repository("owner/repo")
+        
+        # Validate required fields exist
         assert "name" in response
         assert "full_name" in response
+        assert "html_url" in response
+        assert "default_branch" in response
+        
+        # Validate field types
         assert isinstance(response["name"], str)
+        assert isinstance(response["full_name"], str)
+        assert isinstance(response["html_url"], str)
+    
+    def test_api_error_responses(self) -> None:
+        """Ensure API error responses have expected structure."""
+        with pytest.raises(Exception) as exc_info:
+            github_api.get_repository("nonexistent/repo")
+        
+        # Validate error structure
+        assert hasattr(exc_info.value, "status_code")
+        assert exc_info.value.status_code == 404
+
+# tests/contract/test_giturlparse.py
+"""Contract tests for giturlparse library."""
+
+class TestGitURLParseContract:
+    """Validate giturlparse library behavior."""
+    
+    def test_parses_https_github_urls(self) -> None:
+        """Ensure giturlparse handles HTTPS GitHub URLs correctly."""
+        parsed = giturlparse.parse("https://github.com/owner/repo.git")
+        assert parsed.owner == "owner"
+        assert parsed.repo == "repo"
+        assert parsed.platform == "github"
+    
+    def test_parses_ssh_github_urls(self) -> None:
+        """Ensure giturlparse handles SSH GitHub URLs correctly."""
+        parsed = giturlparse.parse("git@github.com:owner/repo.git")
+        assert parsed.owner == "owner"
+        assert parsed.repo == "repo"
 ```
 
-## 🎨 Code Formatting
+#### Contract Test Best Practices
+
+1. **Place in `tests/contract/` directory**: Keep contract tests separate from unit tests
+2. **Test actual library behavior**: Don't mock the external library in contract tests
+3. **Test only what we use**: Only validate the specific features and behaviors our code depends on
+4. **Document expectations**: Clearly document what behavior you're validating and why we depend on it
+5. **Version-specific tests**: Note which library version you're testing against
+6. **Detect breaking changes**: These tests should fail if a library update breaks our assumptions
+7. **Run less frequently**: Contract tests can be slower; mark with `@pytest.mark.contract` if needed
```

Comment:
how will the AI know whether or not marking with `@pytest.mark.contract` is needed? Is it based on how many contract tests there are, since running them is slow?